src/datasets/builder.py

```diff
@@ -351,6 +351,10 @@ def _create_builder_config(self, name=None, custom_features=None, **config_kwarg
         if is_custom:
             logger.warning("Using custom data configuration %s", config_id)
         else:
+            # datasets requiring a manual data dir should set data at init to
+            # pre-instantiated dataset config
+            if builder_config.data_dir is not None:
```

Comment:
@lhoestq -> datasets that require manual data dirs currently can't accept config names. This should resolve it - is the fix ok for you?

Reply:
You shouldn't have to modify the existing configurations at `self.builder_configs[builder_config.name]`
Instead maybe you can change this 
```
if builder_config != self.builder_configs[builder_config.name]:
```
to this
```
if builder_config.equals(self.builder_configs[builder_config.name], ignore_data_dir=True):
```

I can open a PR tomorrow to implement `BuilderConfig.equals` :)

Reply:
Actually this was a bug, I fixed it here:
https://github.com/huggingface/datasets/pull/1932